pandas/_libs/tslibs/strptime.pyx

```diff
@@ -230,6 +273,40 @@ def array_strptime(
         else:
             val = str(val)
 
+        if iso_format:
+            string_to_dts_failed = string_to_dts(
+                val, &dts, &out_bestunit, &out_local,
+                &out_tzoffset, False, fmt, exact
+            )
+            if not string_to_dts_failed:
```

Comment:
What happens if string_to_dts fails?

Reply:
then it continues below

the ISO fast path doesn't accept some single-digit strings for some formats, such as `'202011'` with format `'%Y%m%d'`. So:
- if the format is ISO, try the ISO fast path
- if it fails, let the regular strptime code handle it